CMakeLists.txt

```diff
@@ -596,6 +596,28 @@ if(openPMD_USE_INVASIVE_TESTS)
 endif()
 
 if(BUILD_TESTING)
+    # compile Catch2 implementation part separately
+    add_library(CatchRunner test/CatchRunner.cpp)  # Always MPI_Init with Serial Fallback
+    add_library(CatchMain   test/CatchMain.cpp)    # Serial only
+    target_compile_features(CatchRunner PUBLIC cxx_std_11)
+    target_compile_features(CatchMain   PUBLIC cxx_std_11)
+    set_target_properties(CatchRunner CatchMain PROPERTIES
+        CXX_EXTENSIONS OFF
+        CXX_STANDARD_REQUIRED ON
+        POSITION_INDEPENDENT_CODE ON
+        WINDOWS_EXPORT_ALL_SYMBOLS ON
+    )
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+        target_compile_options(CatchRunner PUBLIC "/bigobj")
+        target_compile_options(CatchMain   PUBLIC "/bigobj")
+    endif()
+    target_link_libraries(CatchRunner PRIVATE openPMD::thirdparty::Catch2)
```

Comment:
If I make those `PUBLIC` I can save the `target_link_libraries(${testname}Tests PRIVATE openPMD::thirdparty::Catch2)` line below